src/m2n/tests/PointToPointCommunicationTest.cpp

```diff
@@ -419,6 +419,276 @@ BOOST_AUTO_TEST_CASE(ConnectionTest, * testing::OnSize(4))
+
+    utils::MasterSlave::_rank       = 0;
+    utils::MasterSlave::_size       = 2;
+    utils::MasterSlave::_masterMode = true;
+    utils::MasterSlave::_slaveMode  = false;
+
+    utils::MasterSlave::_communication->acceptConnection("Fluid.Master", "Fluid.Slave", utils::Parallel::getProcessRank());
+    utils::MasterSlave::_communication->setRankOffset(1);
+
+    Eigen::VectorXd position(dimensions);
+    position <<5.5, 0.0;
+    mesh::Vertex& v1 = mesh->createVertex(position);
+    position << 1.0, 2.0;
+    mesh::Vertex& v2 = mesh->createVertex(position);
+    mesh->createEdge(v1, v2);
+
+    mesh->getConnectedRanks().push_back(0);
+//    mesh->getCommunicationMap()[1].push_back(-1);    
+    
+    break;
+  }
+  case 1: {
+    utils::Parallel::splitCommunicator("Fluid.Slave");
+
+    utils::MasterSlave::_rank       = 1;
+    utils::MasterSlave::_size       = 2;
+    utils::MasterSlave::_masterMode = false;
+    utils::MasterSlave::_slaveMode  = true;
+
+    utils::MasterSlave::_communication->requestConnection("Fluid.Master", "Fluid.Slave", 0, 1);
+
+    Eigen::VectorXd position(dimensions);
+    position <<1.5, 0.0;
+    mesh::Vertex& v1 = mesh->createVertex(position);
+    position << 1.5, 2.0;
+    mesh::Vertex& v2 = mesh->createVertex(position);
+    mesh->createEdge(v1, v2);
+
+    mesh->getConnectedRanks().push_back(1);
+    
+    break;
+  }
+  case 2: {
+    utils::Parallel::splitCommunicator("Solid.Master");
+
+    utils::MasterSlave::_rank       = 0;
+    utils::MasterSlave::_size       = 2;
+    utils::MasterSlave::_masterMode = true;
+    utils::MasterSlave::_slaveMode  = false;
+
+    utils::MasterSlave::_communication->acceptConnection("Solid.Master", "Solid.Slave", utils::Parallel::getProcessRank());
+    utils::MasterSlave::_communication->setRankOffset(1);
+
+    Eigen::VectorXd position(dimensions);
+    position << 1.5, 0.0;
+    mesh::Vertex& v1 = mesh->createVertex(position);
+    position << 1.0, 2.0;
+    mesh::Vertex& v2 = mesh->createVertex(position);
+    mesh->createEdge(v1, v2);
+
+    mesh->getConnectedRanks().push_back(0);
+    //mesh->getCommunicationMap()[1].push_back(-1);     
+
+    break;
+  }
+  case 3: {
+    utils::Parallel::splitCommunicator("Solid.Slave");
+
+    utils::MasterSlave::_rank       = 1;
+    utils::MasterSlave::_size       = 2;
+    utils::MasterSlave::_masterMode = false;
+    utils::MasterSlave::_slaveMode  = true;
+
+    utils::MasterSlave::_communication->requestConnection("Solid.Master", "Solid.Slave", 0, 1);
+
+    Eigen::VectorXd position(dimensions);
+    position <<1.0, 0.0;
+    mesh::Vertex& v1 = mesh->createVertex(position);
+    position << 1.0, 2.0;
+    mesh::Vertex& v2 = mesh->createVertex(position);
+    mesh->createEdge(v1, v2);
+  
+    mesh->getConnectedRanks().push_back(1);
+
+    break;
+  }
+  }
+
+  m2n::PointToPointCommunication c(cf, mesh);
+
+  if (utils::Parallel::getProcessRank() < 2) {
+  
+    c.requestPreConnection("Solid", "Fluid");
```

Comment:
A better unit test design would probably be to not use this functionality here, but to create the pre-connections manually.